### PR TITLE
increase object limits

### PR DESCRIPTION
**Pull request by MjnMixael**
Increases limits as discussed in #SCP. I'm in no big hurry to get this committed, but submitting the PR to ensure the discussion is finished and a decision is made on the best numbers to use.

### CODE REVIEW

**Comment by wookieejedi** [comment]
This increase is very reasonable and would be highly utilized by FotG for one.

**Comment by wookieejedi** [comment]
I also might recommend putting comments after all the bumped sections along with the even perhaps the year?

**Comment by BMagnu** [comment]
IMO, these increases are proper like this. But yeah, the comments for the limits would still need to be edited

**Review by wookieejedi — APPROVED** [review]

File: code/globalincs/globals.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #define SHORT_CALLSIGN_PIXEL_W	80		// max width of short_callsign[] in pixels
 
 // from ship.h
-#define MAX_SHIPS					400			// max number of ship instances there can be.DTP; bumped from 200 to 400
+#define MAX_SHIPS					1000		// max number of ship instances there can be.DTP; bumped from 200 to 400, then to 1000 in 2022
 #define SHIPS_LIMIT					400			// what MAX_SHIPS will be at release time (for error checking in debug mode); dtp Bumped from 200 to 400
```
Comment (EatThePath) [inline code/globalincs/globals.h:36]: do we need to update here too?

**Review by EatThePath — COMMENTED** [review]
Also should we consider bumping the ship class limit for Event Horizon while we're at it?

File: code/globalincs/globals.h (hunk 1 above)
Comment (MjnMixael) [reply code/globalincs/globals.h]: Good catch. I think you're right.

**Comment by notimaginative** [comment]
Just to point out, in case it wasn't considered, this adds nearly 8 MiB onto the stack. The `MAX_SHIPS` bump alone counts for about 5 MiB of that.

**Comment by Goober5000** [comment]
Yeah, these bumps are pretty substantial.  Can we get away with smaller bumps?  IIRC the number tossed around for `MAX_SHIPS` was 500, not 1000.

**Comment by MjnMixael** [comment]
Alright. Set to 500 for ships now.

File: code/globalincs/globals.h [hunk 2]
```diff
@@ -32,8 +32,8 @@
 #define SHORT_CALLSIGN_PIXEL_W	80		// max width of short_callsign[] in pixels
 
 // from ship.h
-#define MAX_SHIPS					400			// max number of ship instances there can be.DTP; bumped from 200 to 400
-#define SHIPS_LIMIT					400			// what MAX_SHIPS will be at release time (for error checking in debug mode); dtp Bumped from 200 to 400
+#define MAX_SHIPS					500		// max number of ship instances there can be.DTP; bumped from 200 to 400, then to 1000 in 2022
+#define SHIPS_LIMIT					500		// what MAX_SHIPS will be at release time (for error checking in debug mode); dtp Bumped from 200 to 400, then to 1000 in 2022
```
Comment (wookieejedi) [inline code/globalincs/globals.h:36]: Comments needs to be updated to 500 instead of 1000 :)

**Review by wookieejedi — CHANGES_REQUESTED** [review]

File: code/globalincs/globals.h (hunk 2 above)
Comment (MjnMixael) [reply code/globalincs/globals.h]: details... details...

**Comment by Goober5000** [comment]
How about 3000 for weapons and 4500 for objects?

**Comment by wookieejedi** [comment]
I might say at least 5000 for objects, for example to account for the higher asteroid increase.

**Comment by MjnMixael** [comment]
Changed to 3000 and 5000 respectively

**Review by wookieejedi — APPROVED** [review]

**Comment by wookieejedi** [comment]
Given the new changes and overall good compromise I'm going to merge this in a day or so (pending any objections).

**Pull request merged**